app/controllers/sessions_controller.rb

```diff
@@ -1,4 +1,34 @@
 class SessionsController < ApplicationController
   def create
+    user_info = request.env['omniauth.auth']
+    name = user_info[:info][:name]
+    uid = user_info[:uid]
+    email = user_info[:info][:email]
+    token = user_info[:credentials][:token]
+    
+    #found_user = User.find_or_create_by(uid: uid, email: email, token: token, name: name)
+
+    found_user = User.find_or_create_by(name: name) do |user|
```

Comment:
Wouldn't we want to find or create by uid? I believe if we're only searching by name we would overwrite the info for an existing user if they happened to have the same name as the new user, whereas the uid would be unique

Reply:
Good point, gonna pivot towards the find method's roots being through UID! Gonna change tests towards that too